projects/storefrontlib/src/cms-components/myaccount/consent-management/components/consent-management.component.ts

```diff
@@ -270,9 +270,8 @@ export class ConsentManagementComponent implements OnInit, OnDestroy {
         if (this.isRequiredConsent(template)) {
           return;
         }
-
-        consentsToGive.push(template);
       }
+      consentsToGive.push(template);
```

Comment:
I don't think this is a good solution. By doing this you are giving consents even if they are already given which returns an error. Since this is Bill's suggestion you should have chat with him to explain the issues with the current APIs.

We can chat more about a solution

<img width="1108" alt="Screen Shot 2020-08-07 at 4 33 25 PM" src="https://user-images.githubusercontent.com/10605724/89687417-39096100-d8ce-11ea-99c6-d2c13f237980.png">